code/src/savefile.c

```diff
@@ -589,6 +589,17 @@ void SaveFile_SetOwnedTradeItemEquipped(void) {
     }
 }
 
+s8 SaveFile_GetIgnoreMaskReactionOption(void) {
+    // This option somehow breaks talking to the Kakariko Mountain Gate guard, so use a workaround
+    if (PLAYER->currentMask == 1 && gGlobalContext->sceneNum == 0x52 &&
+        gSaveContext.linkAge == AGE_CHILD && (gSaveContext.infTable[7] & 0x80) == 0 &&
+        PLAYER->actor.world.pos.x > -230 && PLAYER->actor.world.pos.x < 170 &&
+        PLAYER->actor.world.pos.z > -1498 && PLAYER->actor.world.pos.z < -1098) {
+        return 0;
+    }
```

Comment:
```suggestion
s8 SaveFile_GetIgnoreMaskReactionOption(u32 reactionSet) {
    // This option somehow breaks talking to the Kakariko Mountain Gate guard, so use a workaround
    if (reactionSet == 0x3C &&
        PLAYER->currentMask == 1 && (gSaveContext.infTable[7] & 0x80) == 0) {
        return 0;
    }
```
The DMT Gate guard uses reactionSet 0x3C.

Reply:
This is what I tried at first, but I think the reason it didn't work was because I copied `r1` instead of `r4` because of how it was called in Ghidra. Whoops.